docs/source/user-guide/sql/ddl.md

```diff
@@ -30,9 +90,43 @@ STORED AS PARQUET
 LOCATION '/mnt/nyctaxi/tripdata.parquet';
 ```
 
-CSV data sources can also be registered by executing a `CREATE EXTERNAL TABLE` SQL statement. It is necessary to
-provide schema information for CSV files since DataFusion does not automatically infer the schema when using SQL
-to query CSV files.
+```sql
+CREATE EXTERNAL TABLE test
+    STORED AS CSV
+    WITH HEADER ROW
+    LOCATION 'c:/tmp/test.csv';
+```
+
+Create an external table with partitioned CSV files
+
+```sql
+CREATE EXTERNAL TABLE p_test
+    STORED AS CSV
+    WITH HEADER ROW
+    PARTITIONED BY (year)
+    LOCATION 'c:/tmp/data';
+```
+
+The above statement looks for CSV files in the `c:/tmp/data` directory and creates a table with
+the columns and data types inferred, as well as adding a column for the partition:
+
+TODO: describe rules for inference. which files does it look at, how many rows? is it configurable?
```

Comment:
Currently the default number of rows to scan is defined here:

```
/// default max records to scan to infer the schema
pub const DEFAULT_SCHEMA_INFER_MAX_RECORD: usize = 1000;
```

This is not configurable via the DDL as far as I know.